VimeoNetworking/VimeoNetworking/Models/VIMVideo.m

```diff
@@ -437,7 +437,16 @@ - (BOOL)isRatedMature
 
 - (BOOL)isDRMProtected
 {
-    return self.playRepresentation.drmFiles.fairPlayFile != nil;
+    VIMInteraction *buyInteraction = [self interactionWithName:VIMInteractionNameBuy];
+    BOOL isBuyDRMProtected = buyInteraction.drm;
+    
+    VIMInteraction *rentInteraction = [self interactionWithName:VIMInteractionNameRent];
+    BOOL isRentDRMProtected = rentInteraction.drm;
+    
+    VIMInteraction *subscribeInteraction = [self interactionWithName:VIMInteractionNameSubscribe];
+    BOOL isSubscribeDRMProtected = subscribeInteraction.drm;
+    
+    return isBuyDRMProtected || isRentDRMProtected || isSubscribeDRMProtected;
```

Comment:
I was worried that, because our app hadn't been enabled for DRM, the `drmFiles` array was nil even though we _did_ actually have enough information to determine whether the content was protected by DRM. 

Because the `drm` property isn't suppressed for apps that aren't DRM-enabled we can still get an accurate picture of what is and isn't covered by DRM. I agree it's not ideal though. But I like it more than looking at the `drmFiles` array, which doesn't seem to be as reliable.